### PR TITLE
Minor: Add FixedSizeBinaryTest

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Related to https://github.com/apache/arrow-datafusion/pull/6891

# Rationale for this change

https://github.com/apache/arrow-datafusion/pull/6891 needs some tests, but since there is no fixed size binary examples yet in .slt files I figured I would add some rather than asking @maxburke  to do it

# What changes are included in this PR?

Add basic test coverage for FixedSizeBinary

# Are these changes tested?
Only tests

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/binary.slt [hunk 1]
```diff
@@ -73,3 +73,57 @@ GROUP BY column1;
...
+----
+000102 FixedSizeBinary(3)
+003102 FixedSizeBinary(3)
+NULL FixedSizeBinary(3)
+ff0102 FixedSizeBinary(3)
+000102 FixedSizeBinary(3)
+
+# Comparison
+# https://github.com/apache/arrow-datafusion/pull/6891
+query error DataFusion error: Internal error: Data type FixedSizeBinary\(3, "0,1,2"\) not supported for scalar operation 'eq' on dyn array\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/binary.slt:114]: This should work after https://github.com/apache/arrow-datafusion/pull/6891

**Comment by alamb** [comment]
@maxburke  sadly it seems like your change in https://github.com/apache/arrow-datafusion/pull/6891 isn't sufficient -- the tests in this PR now fail with 

> query error DataFusion error: Arrow error: Compute error: eq_dyn_binary_scalar only supports Binary or LargeBinary arrays

When I try to compare them

**Comment by maxburke** [comment]
> @maxburke sadly it seems like your change in #6891 isn't sufficient -- the tests in this PR now fail with
> 
> > query error DataFusion error: Arrow error: Compute error: eq_dyn_binary_scalar only supports Binary or LargeBinary arrays
> 
> When I try to compare them

Possibly needs this change that was just merged into Arrow? https://github.com/apache/arrow-rs/pull/4492

**Comment by alamb** [comment]
> Possibly needs this change that was just merged into Arrow? https://github.com/apache/arrow-rs/pull/4492

I am pretty sure I did (previously the error message was different -- you can see the error in https://github.com/apache/arrow-datafusion/pull/6895/commits/7fb0d3f744527be3f04f6593231c4be47c3bc87a)

**Comment by alamb** [comment]
I an merging these tests in so we have coverage of the FixedSizeBinary workload and have a place to extend them going forward

**Pull request merged**